solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -1285,6 +1360,47 @@ contract StandardPoolConverter is
         return _amount.mul(_reserveBalance) / _supply;
     }
 
+    /**
+     * @dev returns two samples required for calculating the network fee
+     *
+     * @param currProd  the current product of the balances
+     * @param prevProd  the previous product of the balances
+     *
+     * @return the square root of the current product of the balances
+     * @return the square root of the previous product of the balances
+     */
+    function networkFeePoints(
+        uint256 currProd,
+        uint256 prevProd
+    ) private pure returns (uint256, uint256) {
+        return (
+            currProd > 0 ? MathEx.floorSqrt(currProd) : 0,
+            prevProd > 0 ? MathEx.floorSqrt(prevProd) : 0
+        );
+    }
+
+    /**
+     * @dev returns the network fee ratio
+     *
+     * @param currPoint     the current product of the balances
+     * @param prevPoint     the previous product of the balances
+     * @param networkFee    the relative portion (in PPM units) that
+     * should be taken from the conversion fees accumulated in this pool
+     *
+     * @return numerator of the portion that should be taken from the balances of this pool
+     * @return denominator of the portion that should be taken from the balances of this pool
+     */
+    function networkFeeRatio(
+        uint256 currPoint,
+        uint256 prevPoint,
+        uint256 networkFee
+    ) private pure returns (uint256, uint256) {
+        return (
+            (currPoint - prevPoint) * networkFee,
+            (currPoint * (PPM_RESOLUTION - networkFee)).add(prevPoint * networkFee * 2)
```

Comment:
Can you elaborate on this calculation? Why are we taking `1 - networkFee`? Multiplying by 2?

Reply:
Uniswap's code:
```
uint numerator = totalSupply.mul(rootK.sub(rootKLast));
uint denominator = rootK.mul(5).add(rootKLast);
```

Translation Phase #1:
```
uint n = totalSupply * (currPoint - prevPoint);
uint d = currPoint * 5 + prevPoint;
```

Translation Phase #2:
```
uint n = totalSupply * (currPoint - prevPoint);
uint d = currPoint * (PPM_RESOLUTION / networkFee - 1) + prevPoint;
```

Translation Phase #3:
```
uint n = totalSupply * (currPoint - prevPoint);
uint d = currPoint * (PPM_RESOLUTION - networkFee) / networkFee + prevPoint;
```

Translation Phase #4:
```
uint n = totalSupply * (currPoint - prevPoint);
uint d = (currPoint * (PPM_RESOLUTION - networkFee) + prevPoint * networkFee) / networkFee;
```

Translation Phase #5:
```
uint n = totalSupply * (currPoint - prevPoint) * networkFee;
uint d = currPoint * (PPM_RESOLUTION - networkFee) + prevPoint * networkFee;
```

This gives the fee in units of the pool tokens.
We need to convert half of it to units of reserve token 1, and half of it to units of reserve token 2.

In order to represent half, we multiply the denominator by 2:
```
uint d = currPoint * (PPM_RESOLUTION - networkFee) + prevPoint * networkFee * 2;
```

In order to represent by units of reserve token X, we:
1. Multiply by the balance of the pool in reserve token X
2. Divide by the total supply of pool tokens

**Important Note:**
Note that this relies on spot rate.
In order to rely on the bonding-curve rate, we would need to convert this amount using `removeLiquidityReturn`.
We may want to dwell on this in order to determine which one of these two options we should be using.

In any case, multiplying by the balance of the pool in reserve token X is conducted outside of this function.
In this function, we only divide by the total supply of pool tokens, so this:
```
uint n = totalSupply * (currPoint - prevPoint) * networkFee;
```
Changes to this:
```
uint n = (currPoint - prevPoint) * networkFee;
```

Reply:
👍 👍 👍